src/Symfony/Component/Console/Application.php

```diff
@@ -535,10 +535,11 @@ public function addCommands(array $commands): void
      * If a command with the same name already exists, it will be overridden.
      * If the command is not enabled it will not be added.
      */
-    public function add(Command $command): ?Command
+    public function add(object $command): ?Command
```

Comment:
We added the new `addCommand` method in https://github.com/symfony/symfony/pull/60394 to preserve backward compatibility; as [changing the parameter type is a breaking change](https://symfony.com/doc/current/contributing/code/bc.html#changing-classes).

This must be reverted.